### ISSUE TITLE
Which of the methods are coroutine-safe?

### ISSUE THREAD

**Issue by yifeikong** [mosquito/aiofile#81]
Can I share a file/reader/writer object in different coroutines without Lock? Which methods are coroutine-safe?

**Comment by mosquito** [comment]
TL/DR:  Writer - Yes, Reader - Yes, LineReader - Probably yes. However, the point is not quite clear, since these are objects that just contain a pointer to an offset, they are cheaper to copy and send each copy to the diferrent coroutines in my opinion.

I think questions like this has to do with some misunderstanding of exactly how this library designed under the hood. It's probably my bad, fail to explain this in the docs.

There is `AIOFile`, which is like a low-level abstraction that owns an instance of one of the implementations of `AsyncioContext`. It is a very simple object that can read and write **only by offset**, for that you get absolute competitiveness when reading and writing, without having to block anything.

So, objects of class `BinaryFileWrapper` and `TextFileWrapper` which is what the `async_open` function returns and mimicking the normal `open` in python. They have a caret that points to the file offset, and of course there are locks for read and write that do not allow a race between the operations and the value in the caret. The dark side of this simplicity is the lost read and write concirrency.

However, the library has some helpers that allow you to use one `AIOFile` concurrently. If you imagine that the user code has different coroutine roles which have different responsibilities, for example one writes data, the others only reads, then you can keep one `AIOFile` object and pass it wrapped in objects `Writer`, `Reader` and `LineReader` to this coroutines.

This is not the first question in this vein, I do not know how best to reflect this in the documentation, it would be great if someone could help me.

**Comment by yifeikong** [comment]
Thanks for the timely reply! After some investigation, I'm using `caio` with Lock now, which suits me better. Great work for these two libraries!
